check-config-win.py

```diff
@@ -1,64 +1,123 @@
 found_key = False
-
-for root in key_roots:
-    key = winreg.OpenKey(getattr(winreg, root), key_path)
+print("- Checking Registry:")
+for k in HKEYS:
     try:
-        print("Checking:", root, key_path)
-        res = winreg.QueryValueEx(key, "")
+        print(fr"{k}\{FF2MPV_KEY} ... ", end="")
+        key_open = winreg.OpenKey(HKEYS[k], FF2MPV_KEY)
+        hkey_found = HKEYS[k]
+        print("Found.")
     except FileNotFoundError:
-        print("...error finding key")
+        print("Not found.")
+        error = True
         continue
-
+    error = False
     found_key = True
     break
 
 if not found_key:
-    raise ValueError("Could not find a registry entry, aborting.")
+    if args.install:
+        # The intermediate missing key are also created.
+        key_open = winreg.CreateKey(HKEYS["HKEY_CURRENT_USER"], FF2MPV_KEY)
+        print("Key created.")
 
-json_path = res[0]
-print("Path from registry key is:", json_path)
-if not os.path.exists(json_path):
-    raise ValueError("JSON file does not exist:", json_path)
+if not args.uninstall:
+    ff2mpv_value = winreg.QueryValue(key_open, "")
+    if args.install:
+        if ff2mpv_value != FF2MPV_JSON:
+            winreg.SetValue(
+                HKEYS["HKEY_CURRENT_USER"], FF2MPV_KEY, winreg.REG_SZ, FF2MPV_JSON
+            )
+            ff2mpv_value = winreg.QueryValue(key_open, "")
+            print("Value set/updated.\nRestart Firefox if it was running.")
+        else:
+            print("Nothing to update.")
 
-try:
-    bat_data = json.load(open(json_path, "r"))
-except json.decoder.JSONDecodeError:
-    raise ValueError("Parsing error. Is {} a JSON file?".format(json_path))
+    else:
+        if ff2mpv_value != "":
+            print("Value of the key is:", ff2mpv_value)
+            try:
+                os.path.exists(ff2mpv_value)
+                try:
+                    json.load(open(ff2mpv_value, "r"))
+                except json.decoder.JSONDecodeError:
+                    print(f"error: Is {os.path.basename(ff2mpv_value)} a JSON file?")
+            except FileNotFoundError:
+                print("error: The file does not exist.")
+                error = True
+        else:
+            print("Empty value in the key.")
 
-bat_path = bat_data["path"]
-print("Path from JSON is:", bat_path)
-if not os.path.exists(bat_path):
-    raise ValueError(".bat does not exist:", bat_path)
-
-py_lines = open(bat_path, "r").readlines()
-py_path = None
-for line in py_lines:
-    if line.startswith("call python "):
-        py_path = line[12:].replace("\\\\", "\\").strip()
-
-if not py_path:
-    raise ValueError("No python script in the batch file.")
+    print('- Environment variable "Path":')
+    try:
+        subprocess.run("mpv --version", check=False)
+    except FileNotFoundError:
+        print("error: Path for mpv missing.")
+        print(
+            '\nPress Win + R, then type or copy/paste "rundll32.exe sysdm.cpl,EditEnvironmentVariables".'
```

Comment:
I'm not sure if this is the best way to edit environment variables? If you press Start and type in "Environment Variables", you get redirected to the page, without having to type in commands into the Windows runner

Reply:
`'\nPress Win (key between Ctrl and Alt), then type "Environment Variables".' `
looks good ?